internal/node/node.go

```diff
@@ -130,3 +125,28 @@ func removeLabels(node *v1.Node, labels []string) {
 		)
 	}
 }
+
+func getLabelsToRemove(node *v1.Node, nmcObj *v1beta1.NodeModulesConfig) []string {
+	var labelsToRemove []string
+	for _, module := range nmcObj.Spec.Modules {
+		toDeleteLabel := len(node.Spec.Taints) == 0
```

Comment:
Doesn't it mean that we will delete the label if the node has no taints? If the node has no taints then the module will be loaded and we don't need to remove the label.